src/other/index.jl

```diff
@@ -113,11 +113,20 @@ Base.getindex(x::AbstractIndex, idx::Real) = Int(idx)
 Base.getindex(x::AbstractIndex, idx::AbstractVector{Union{Bool, Missing}}) =
     getindex(x, collect(Missings.replace(idx, false)))
 Base.getindex(x::AbstractIndex, idx::AbstractVector{Bool}) = find(idx)
-Base.getindex(x::AbstractIndex, idx::AbstractVector{T}) where {T >: Missing} =
-    getindex(x, collect(skipmissing(idx)))
 Base.getindex(x::AbstractIndex, idx::AbstractRange) = [idx;]
-Base.getindex(x::AbstractIndex, idx::AbstractVector{T}) where {T <: Real} = convert(Vector{Int}, idx)
+Base.getindex(x::AbstractIndex, idx::AbstractVector{<:Union{Real, Missing}}) =
+    Int[i for i in skipmissing(idx)]
+Base.getindex(x::AbstractIndex, idx::AbstractVector{<:Real}) = convert(Vector{Int}, idx)
+Base.getindex(x::AbstractIndex, idx::AbstractVector{Union{Symbol, Missing}}) =
+    [x.lookup[i] for i in skipmissing(idx)]
 Base.getindex(x::AbstractIndex, idx::AbstractVector{Symbol}) = [x.lookup[i] for i in idx]
+function Base.getindex(x::AbstractIndex, idx::AbstractVector)
+    idxs = skipmissing(idx)
+    try # allow either vector of numbers
```

Comment:
I don't really like that approach. It would be simpler to do `[convert(Union{Int, Symbol}, i) for i in idxs]`. If we really don't want to allow mixing integers and symbols, we can always check the type of the result at the end.

Reply:
fixed